chainerx_cc/chainerx/python/array.h

```diff
@@ -22,6 +24,8 @@ using ConstArrayBodyPtr = std::shared_ptr<const ArrayBody>;
 
 ArrayBodyPtr MakeArray(pybind11::handle object, pybind11::handle dtype, bool copy, pybind11::handle device);
 
+ArrayBodyPtr MakeArray(pybind11::handle object, nonstd::optional<Dtype>& dtype, bool copy, Device& device);
```

Comment:
Can you take the dtype by `const &` ?
```suggestion
ArrayBodyPtr MakeArray(pybind11::handle object, const nonstd::optional<Dtype>& dtype, bool copy, Device& device);
```